claat/nodes/survey_test.go

```diff
@@ -0,0 +1,210 @@
+package nodes
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestNewSurveyNode(t *testing.T) {
+	tests := []struct {
+		name     string
+		inID     string
+		inGroups []*SurveyGroup
+		out      *SurveyNode
+	}{
+		{
+			name: "Empty",
+			out: &SurveyNode{
+				node: node{typ: NodeSurvey},
+			},
+		},
+		{
+			// TODO: should the absence of an ID be an error?
```

Comment:
Good question, I think it should either evaluate as empty or as a valid depending on the desired behavior for SurveyNode.

Reply:
Yeah, we need to figure out what the desired behavior should be. I would like to defer this to a separate PR

Reply:
SGTM